client/landing/stepper/declarative-flow/internals/steps-repository/domains/index.tsx

```diff
@@ -267,7 +267,8 @@ const DomainsStep: Step = function DomainsStep( { navigation, flow } ) {
 	const onUseYourDomainClick = ( domain?: string ) => {
 		if ( domain && isHundredYearDomainFlow( flow ) ) {
 			const leaveFlowFunction = exitFlow ?? window.location.assign;
-			leaveFlowFunction( `/setup/hundred-year-domain-transfer?new=${ domain }` );
+			leaveFlowFunction( `/setup/hundred-year-domain-transfer/domains?new=${ domain }` );
```

Comment:
Fixes the second minor bug described in this PR - WordPress.com logo blinkling:

This happens because the user gets redirected from no step (`/`) to the component's first step, (`/domains`). 
If we send them directly to this step, no redirection will be needed, thus we won't get the logo blinking.